### DISCUSSION TITLE
Disable the copy to clipboard button from a single code chunk

### DISCUSSION THREAD

**Discussion by Qepanna** [quarto-dev/quarto-cli#7885] [Q&A] [answered]
### Description

Is it possible to disable the copy to clipboard button on a single code chunk?

I know we can disable the button in the whole document this way:
```
format:
  html:
    code-copy: false
```

But I'm looking to disable it on a few selected code chunks only.

**Comment by cscheid** [comment]
Not at the moment. We haven't had requests for it until yours. What's your use case?

**Comment by Qepanna** [comment]
I am making a webpage to teach unexperienced students how to use an R package. In my description of the package, I include the general "Usage" of the package, which is a piece of code that is not intended to be copied (they should copy the examples, not the Usage).

For the moment, I included a callout-note that says they should not copy the Usage, but since I have that Usage section comming up dozens of times, I would like to avoid putting up that note dozens of times as well.

**Comment by cderv** [comment, marked as answer]
You can easily leverage CSS for that, by placing an attributes on the div for the code

````markdown
---
title: "test"
format: 
  html:
    theme: 
      - custom.scss
---

```{.r code-copy=false}
# do not copy
1 + x
```


```{r}
#| eval: false
#| code-copy: false
# do not copy
1 + x
```
````

the SCSS file being 
````scss
/*-- scss:defaults --*/

/*-- scss:rules --*/
div[data-code-copy="false"] button.code-copy-button {
  display: none;
}
````

In the .qmd

* The first code block leverage non-executable code block syntax which allows class and attributes (through [Pandoc feature](https://pandoc.org/MANUAL.html#extension-fenced_code_attributes))
* The second code block is executable, and leverage the fact that unknown option are passed asis to the wrapping div. 

You can also use custom div to wrap the whole block 
````markdown
::: {code-copy='false'}
...
:::
````

And target this Div in CSS. 

You can also use your current document structure (like the only code block below `Usage` header section ) to target the code block to not display the button. 

Hope it helps

**Reply by Qepanna** [reply]
I used your solution like this:

```{r}
#| eval: false
#| code-copy: false
# do not copy
1 + x
```
With the .scss file and it worked. Thank you :)
